Nitric.Sdk/Document/DocumentRef.cs

```diff
@@ -44,25 +47,38 @@ public Document<T> Get()
             DocumentGetRequest request = new DocumentGetRequest();
             request.Key = this.Key.ToKey();
 
-            var response = this.documentClient.Get(request);
-
-            return new Document<T>(
-                this,
-                DocumentToGeneric(response.Document.Content)
-            );
+            try
+            {
+                var response = this.documentClient.Get(request);
+                return new Document<T>(
+                    this,
+                    DocumentToGeneric(response.Document.Content)
+                );
+            }
+            catch (RpcException re)
+            {
+                throw NitricException.Exceptions[re.StatusCode](re.Message);
```

Comment:
Is it possible for the StatusCode to be missing from the dictionary (now or in future). If so, this should probably be wrapped in a function that returns and UnknownException if the status code is unexpected.

Reply:
Then reuse the function everywhere these exceptions are being constructed

Reply:
Done.